source/common/crypto/utility.cc

```diff
@@ -41,6 +43,64 @@ std::vector<uint8_t> Utility::getSha256Hmac(const std::vector<uint8_t>& key,
   return hmac;
 }
 
+const VerificationOutput Utility::verifySignature(const absl::string_view& hash, void* ptr,
+                                                  const std::vector<uint8_t>& signature,
+                                                  const std::vector<uint8_t>& text) {
+  // Step 1: get public key
+  auto pub_key = reinterpret_cast<EVP_PKEY*>(ptr);
```

Comment:
I think the type signature should be `EVP_PKEY`, interfaces are best typed.

Reply:
Do you mean change the `void*` to `EVP_PKEY*`?